ClosedXML/Excel/Ranges/XLRangeAddress.cs

```diff
@@ -203,22 +252,46 @@ public string ToString(XLReferenceStyle referenceStyle, bool includeSheet)
 
         public override bool Equals(object obj)
         {
-            var other = obj as XLRangeAddress;
-            if (other == null)
+            if (!(obj is XLRangeAddress))
+            {
                 return false;
-            return Worksheet.Equals(other.Worksheet)
-                   && FirstAddress.Equals(other.FirstAddress)
-                   && LastAddress.Equals(other.LastAddress);
+            }
+
+            var address = (XLRangeAddress)obj;
+            return _firstAddress.Equals(address._firstAddress) &&
+                   _lastAddress.Equals(address._lastAddress) &&
+                   EqualityComparer<XLWorksheet>.Default.Equals(Worksheet, address.Worksheet);
         }
 
         public override int GetHashCode()
         {
-            return
-                Worksheet.GetHashCode()
-                ^ FirstAddress.GetHashCode()
-                ^ LastAddress.GetHashCode();
+            var hashCode = -778064135;
```

Comment:
`GetHashCode` should not reference mutable fields. Change `_firstAddress` and `_lastAddress` to `FirstAddress` and `LastAddress`

Reply:
Here I had a short discussion on this matter. https://www.jetbrains.com/help/resharper/2017.3/NonReadonlyMemberInGetHashCode.html#comment-3862369295 Am I wrong?

Reply:
I made `Worksheet` read-only anyway.

Reply:
You're right. I forgot it's a struct. But this change helps for the static analysis tools to stop complaining and simplifies the code. And it simplifies the code. Thanks